src/main/java/seedu/address/storage/JsonAdaptedRecord.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.storage;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.record.Condition;
+import seedu.address.model.record.DateTime;
+import seedu.address.model.record.Record;
+
+/**
+ * Jackson-friendly version of {@link Record}.
+ */
+public class JsonAdaptedRecord {
+
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("ddMMyyyy HHmm");
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Person's %s field is missing!";
+    private final LocalDateTime dateTime;
+    private final List<JsonAdaptedCondition> conditions = new ArrayList<>();
+
+
+    /**
+     * Constructs a {@code JsonAdoptedRecord} with the given record details.
+     */
+    @JsonCreator
+    public JsonAdaptedRecord(@JsonProperty("dateTime") LocalDateTime dateTime,
+                             @JsonProperty("condition") List<JsonAdaptedCondition> conditions) {
+        this.dateTime = dateTime;
+        if (conditions != null) {
```

Comment:
Instantiating a new ArrayList and checking if it is not null is not necessary i think. This means that if the Record condition is null, A JSON with a empty condition is created which shouldnt be the case, it should be an empty Record.

Reply:
Null check is for the the parameter, not the attribute of the JsonAdaptedRecord. Therefore, I think I need the check to pass the test cases.